rasa/core/tracker_store.py

```diff
@@ -104,7 +105,7 @@ def create(
         import sqlalchemy.exc
 
         try:
-            return _create_from_endpoint_config(obj, domain, event_broker)
+            return AwaitableTrackerStore(_create_from_endpoint_config(obj, domain, event_broker))
```

Comment:
Can we skip wrapping if the tracker store already supports `async`?

Reply:
Hi @joejuzl ! I was thinking if someone has a custom tracker store defined in an endpoint config file which doesn't currently have async methods, this would handle that case

Reply:
I meant could we check if the tracker store returned from `_create_from_endpoint_config` had async methods, and only wrap if not. This way we could also raise a deprecation warning when a non-async tracker store is used.

Reply:
I like that. Done!